### ISSUE TITLE
color-contrast: oklch()/oklab()/color() backgrounds are unparsed, walked past, and resolved to white

### ISSUE THREAD

**Issue by ckundo** [AccessLint/accesslint#23]
Split out of #21 / #22, which fixed a different root cause with the same symptom.

## Observed

Chromium does **not** convert modern color functions to `rgb()` in computed styles. Verified in a browser test:

```
color .a { color: oklch(0.5 0.1 250) }              → "oklch(0.5 0.1 250)"
color .b { color: color-mix(in srgb, red 50%, blue) } → "color(srgb 0.5 0 0.5)"
color .c { color: rgb(0 0 0 / 50%) }                  → "rgba(0, 0, 0, 0.5)"   ← legacy, fine
```

`parseColor` in `core/src/rules/utils/color.ts` handles named colors, hex, and `rgb()`/`rgba()` in both comma and space syntax. It returns `null` for `oklch()`, `oklab()`, and `color(srgb ...)`.

## Two consequences

**False positive (same shape as #21).** In `_computeEffectiveBg`, an unparseable `background-color` falls through to `current = current.parentElement; continue;` — the walk steps *past* the real background and can reach the root default of white. Dark `oklch()` surfaces with light text get judged against white and flagged.

**False negative.** In `checkContrast`, `parseColor(style.color)` returning `null` hits `continue`, silently skipping the element. Any text colored with `oklch()` is never checked at all.

Both also apply to `parseGradientStops`: an `oklch()` gradient yields zero stops, so `checkGradientContrast` returns `null` and the element is skipped.

## Why this likely matters now

Tailwind v4 emits `oklch()` for its entire default palette, so `bg-*` and `text-*` utilities produce computed values this engine cannot read. The page reported in #21 may well be hitting this in addition to the gradient bug fixed in #22.

## Suggested fix

Extend `parseColor` with conversion for `oklch()` / `oklab()` (polar → cartesian → linear sRGB matrix → gamma) and `color(srgb r g b)`. `lab()`/`lch()` need an extra XYZ D50 hop and could follow later.

Prefer conversion over a bail-out: making unparseable backgrounds return `null` from `_computeEffectiveBg` would kill the false positive but turn every oklch page into a silent skip.

## Tests

Must be browser tests (`core/src/integration/color-contrast.browser.test.ts`) — happy-dom's computed style won't reproduce this. Cover: dark `oklch()` background with light text (currently flagged, should pass), `oklch()` text color (currently skipped, should be evaluated), and an `oklch()` gradient.
